### ISSUE TITLE
JP2Gen TRACE JP2 files are badly formed, as seen by SunPy

### ISSUE THREAD

**Issue by wafels** [Helioviewer-Project/jp2gen#31]
As noted in #28, the TRACE JP2 pipeline produces files which have the wrong dimensions

```
In [2]: m = sunpy.map.Map("/Users/ireland/2008_11_12__17_27_54_000__TRACE_TRACE_TRACE_171.jp2")
WARNING: SunpyUserWarning: This file contains more than 2 dimensions. Data will be truncated to the first two dimensions. [sunpy.map.mapbase]

In [6]: m.data.shape
Out[6]: (1024, 3)
In [7]: m.dimensions
Out[7]: PixelPair(x=<Quantity 3. pix>, y=<Quantity 1024. pix>)
```

I think the issue lies here: https://github.com/Helioviewer-Project/jp2gen/blob/master/idl/trace/hvs_trace.pro#L60 . The details structure gets the palette keyword. The value is an array of [3, 256]. The code that writes the JP2 file contains an option to use the palette in the details structure: https://github.com/Helioviewer-Project/jp2gen/blob/master/idl/gen/hv_write_jp2_lwg.pro#L581 . I think what is happening is that IDL then uses the palette to write out a JP2 file with RGB channels. I'll post this as an issue.

**Comment by wafels** [comment]
The fix seems to have worked - see attached.
![Figure_1](https://user-images.githubusercontent.com/983575/154764197-ae3bbf97-dc0e-4d40-b547-89bbf852d3c2.png)

**Comment by wafels** [comment]
Fixed by #32. Closing
